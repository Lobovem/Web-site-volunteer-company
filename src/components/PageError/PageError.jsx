import { Link } from 'react-router-dom';
import { Btn } from '../kit/Btn/Btn';
import s from './PageError.module.scss';

export const PageError = () => {
  return (
    <div className={s.pageError}>
      <div className={s.pageError__wrap}>
        <h1 className={s.pageError__title}>
          <span>404</span>
        </h1>
        <p className={s.pageError__desc}>Упс! Щось пішло не так!</p>
        <Link to="/">
          <Btn title="На головну" className="btn" type="button" />
        </Link>
      </div>
      <div className={s.pageError__footerWrap}>
        <div className={s.pageError__line}></div>
      </div>
    </div>
  );
};
