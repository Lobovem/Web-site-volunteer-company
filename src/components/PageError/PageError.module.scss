@import './src/styles/MixinsMedia.module.scss';
@import './src/styles/Variables.module.scss';
@import './src/styles/MixinsLayout.module.scss';
@import './src/styles/MixinsKit.module.scss';

.pageError {
  &__wrap {
    @include layoutMain;
    display: flex;
    justify-content: center;
    flex-direction: column;
    align-items: center;
    gap: 25px;
    padding: 110px 0;

    @include lg-min {
      gap: 80px
    }
  }

  &__title {
    font-weight: $font-weight-500;
    font-size: $font-size-128;
    line-height: 128px;
    mix-blend-mode: normal;
    mix-blend-mode: normal;
    background: $gradient-main;
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;

    @include lg-min {
      font-size: $font-size-300;
      line-height: 300px;
    }
  }

  &__desc {
    font-size: $font-size-20;
    font-weight: $font-weight-500;
    line-height: 20px;
    text-transform: uppercase;
    color: $color-black;
    margin-bottom: 15px;

    @include lg-min {
      font-size: $font-size-24;
      line-height: 24px;
    }
  }

  &__footerWrap {
    background: $color-grey;
    height: 125px;

    @include lg-min {
      height: 275px;
    }

  }

  &__line {
    height: 30px;
    background: $gradient-main;
  }
}