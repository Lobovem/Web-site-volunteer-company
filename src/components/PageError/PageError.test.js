import { render, screen } from '@testing-library/react';
import { PageError } from '../PageError/PageError';
import { MemoryRouter } from 'react-router-dom';

describe('PageError', () => {
  test('renders PageError component', () => {
    const { container } = render(
      <MemoryRouter>
        <PageError />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });

  test('search title tag h1 with text 404', () => {
    render(
      <MemoryRouter>
        <PageError />
      </MemoryRouter>
    );
    const h1Element = screen.getByRole('heading', { name: /404/i });
    expect(h1Element).toBeInTheDocument();
  });

  test('renders and search component button', () => {
    render(
      <MemoryRouter>
        <PageError />
      </MemoryRouter>
    );
    const buttonElement = screen.getByRole('button', { type: 'button' });
    expect(buttonElement).toBeInTheDocument();
  });
});
