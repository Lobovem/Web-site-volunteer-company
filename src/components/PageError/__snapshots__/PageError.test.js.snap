// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageError renders PageError component 1`] = `
<div>
  <div
    class="pageError"
  >
    <div
      class="pageError__wrap"
    >
      <h1
        class="pageError__title"
      >
        <span>
          404
        </span>
      </h1>
      <p
        class="pageError__desc"
      >
        Упс! Щось пішло не так!
      </p>
      <a
        href="/"
      >
        <button
          class="btn"
          type="button"
        >
          На головну
        </button>
      </a>
    </div>
    <div
      class="pageError__footerWrap"
    >
      <div
        class="pageError__line"
      />
    </div>
  </div>
</div>
`;
